cmd/gardener-scheduler/app/gardener_scheduler.go

```diff
@@ -111,175 +77,112 @@ func NewCommandStartGardenerScheduler() *cobra.Command {
 			if err := opts.validate(args); err != nil {
 				return err
 			}
-			return opts.run(cmd.Context())
+
+			return runCommand(cmd, opts)
 		},
 	}
 
 	flags := cmd.Flags()
 	verflag.AddFlags(flags)
 	opts.AddFlags(flags)
-	return cmd
-}
 
-// GardenerScheduler represents all the parameters required to start the
-// Gardener scheduler.
-type GardenerScheduler struct {
-	Config            *config.SchedulerConfiguration
-	Identity          *gardencorev1beta1.Gardener
-	GardenerNamespace string
-	ClientMap         clientmap.ClientMap
-	Logger            *logrus.Logger
-	Recorder          record.EventRecorder
-	LeaderElection    *leaderelection.LeaderElectionConfig
+	return cmd
 }
 
-// NewGardenerScheduler is the main entry point of instantiating a new Gardener Scheduler.
-func NewGardenerScheduler(ctx context.Context, cfg *config.SchedulerConfiguration) (*GardenerScheduler, error) {
-	// validate the configuration
-	if err := validation.ValidateConfiguration(cfg); err != nil {
-		return nil, err
-	}
-
-	// Initialize logger
-	logger := logger.NewLogger(cfg.LogLevel)
-	logger.Info("Starting Gardener scheduler ...")
-	logger.Infof("Feature Gates: %s", schedulerfeatures.FeatureGate.String())
-
-	// Prepare a Kubernetes client object for the Garden cluster which contains all the Clientsets
-	// that can be used to access the Kubernetes API.
-	if kubeconfig := os.Getenv("KUBECONFIG"); kubeconfig != "" {
-		cfg.ClientConnection.Kubeconfig = kubeconfig
+func runCommand(cmd *cobra.Command, opts *Options) error {
```

Comment:
Only the command's context is needed.

```suggestion
func runCommand(ctx context.Context, opts *Options) error {
```